src/Routing/PageSlugConditionChecker.php

```diff
@@ -35,17 +36,24 @@ final class PageSlugConditionChecker
      */
     private $localeContext;
 
+    /**
+     * @var DateTimeProviderInterface
+     */
+    private $dateTimeProvider;
+
     /**
      * PageSlugConditionChecker constructor.
      */
     public function __construct(
         PageRepositoryInterface $pageRepository,
         ChannelContextInterface $channelContext,
-        LocaleContextInterface $localeContext
+        LocaleContextInterface $localeContext,
+        DateTimeProviderInterface $dateTimeProvider
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
**💣 chore:** Let's put this php 8 style. This version does not support php 7.4 anyway ‏

Reply:
It could be a global refactoring could be better